### DISCUSSION TITLE
How are user textures supposed to be handled in 0.16?

### DISCUSSION THREAD

**Discussion by Jengamon** [emilk/egui#1015] [Q&A] [answered]
The rename of Texture -> FontImage kinda threw me off, and I want to try to update egui_wgpu_backend (as I use wgpu) to support 0.16.

**Comment by emilk** [comment, marked as answer]
The rename is just a rename, not much should change: https://github.com/emilk/egui/commit/190c85a40ff022f8f3bb3133b90d17e4880ecbd8#diff-0660ea948cba6068c6534d1833d321f20e395448a1d4cd6c5ca9dba90ad3a023

For user textures, check out the diff for `egui_glium/src/painter.rs`: https://github.com/emilk/egui/pull/999/files#diff-23f23e44bac11e24b5797a2fa0734dbcd65b21830492cde1ad030e2314e76e7f

**Reply by Jengamon** [reply]
Ok, what's the difference between `epi::TextureAllocator` and `epi::NativeTexture`?

**Reply by emilk** [reply]
`TextureAllocator` is for loading images into textures. `NativeTexture` is for when you have a native texture (e.g. `glow` texture if you are using `egui_glow`) and want to get an `egui::TextureId` for is so you can display in egui.
